src/Sentry.Unity/SentryOptionsUtility.cs

```diff
@@ -9,35 +9,105 @@ public static void SetDefaults(SentryUnityOptions options, IApplication? applica
         {
             application ??= ApplicationAdapter.Instance;
 
-            // 'Optimal' and 'Fastest' don't work on IL2CPP. Forcing 'NoCompression'.
-            options.RequestBodyCompressionLevel = CompressionLevelWithAuto.NoCompression;
-            options.IsEnvironmentUser = false;
+            options.Enabled = Enabled;
+            options.Dsn = string.Empty;
+            options.CaptureInEditor = CaptureInEditor;
+            options.RequestBodyCompressionLevel = RequestBodyCompressionLevel;
+            options.AttachStacktrace = AttackStackTrace;
+            options.StackTraceMode = StackTraceMode;
+            options.SampleRate = SampleRate;
+            options.IsEnvironmentUser = IsEnvironmentUser;
 
-            SetRelease(options, application);
-            SetEnvironment(options, application);
-            SetCacheDirectoryPath(options, application);
+            options.Release = Release(application);
+            options.Environment = Environment(application);
+
+            options.CacheDirectoryPath = application.PersistentDataPath;
+
+            options.Debug = Debug;
+            options.DebugOnlyInEditor = DebugOnlyInEditor;
+            options.DiagnosticLevel = DiagnosticLevel;
+
+            TryAttachLogger(options, application);
         }
 
-        private static void SetRelease(SentryUnityOptions options, IApplication application)
+        public static void SetDefaults(ScriptableSentryUnityOptions options)
         {
-            options.Release ??= application.ProductName is string productName
+            options.Enabled = Enabled;
+            options.Dsn = string.Empty;
+            options.CaptureInEditor = CaptureInEditor;
+            options.Debug = Debug;
+            options.DiagnosticLevel = DiagnosticLevel;
+            options.AttachStacktrace = AttackStackTrace;
+            options.SampleRate = SampleRate;
+
+            options.ReleaseOverride = string.Empty;
+            options.EnvironmentOverride = string.Empty;
+
+            options.EnableOfflineCaching = true;
+
+            options.Debug = Debug;
+            options.DebugOnlyInEditor = DebugOnlyInEditor;
+            options.DiagnosticLevel = DiagnosticLevel;
+        }
+
+        private static bool Enabled = true;
+        private static bool CaptureInEditor = true;
```

Comment:
Not sure I understand the need for separate variables for default values. Why not setting them directly?

Reply:
I think it'll be easier to read to just set all defaults straight in the method instead of declaring these static fields.

A single method where we can read through and easily spot all the default values